SWLOR.Game.Server/Service/BaseService.cs

```diff
@@ -922,6 +922,96 @@ public int CalculateResourceCapacity(Guid pcBaseID)
             return (int) (resourceMax + resourceMax * siloBonus);
         }
 
+        public string UpgradeControlTower(NWCreature user, NWItem item, NWObject target)
+        {
+            //--------------------------------------------------------------------------
+            // Called when a PC uses a structure item on a placed object.
+            // Returns "" if either the item or the target are not control towers.
+            // Returns a success or failure message otherwise.
+            //--------------------------------------------------------------------------
+            NWPlayer player = user.Object;
+
+            //--------------------------------------------------------------------------
+            // Check that the item is a control tower.
+            //--------------------------------------------------------------------------
+            int newTowerStructureID = item.GetLocalInt("BASE_STRUCTURE_ID");
+            BaseStructure newTower = _data.Single<BaseStructure>(x => x.ID == newTowerStructureID);
+            if (newTower.BaseStructureTypeID != (int)BaseStructureType.ControlTower)
+            {
+                return "";
+            }
+
+            //--------------------------------------------------------------------------
+            // Check that the target is a control tower.
+            //--------------------------------------------------------------------------
+            string sTowerID = target.GetLocalString("PC_BASE_STRUCTURE_ID");
+            if (string.IsNullOrWhiteSpace(sTowerID)) return "";
+            Guid towerGuid;
+
+            try
+            {
+                towerGuid = new Guid(sTowerID);
+            }
+            catch (Exception e)
+            {
+                Console.Write("Failed to convert GUID: " + sTowerID);
```

Comment:
Can you use the ErrorService here so that it'll get pumped into the DB log? This'll let us check the error log just in case someone isn't monitoring the server when it happens